### PR TITLE
utils_misc: fix decimals conversion between differen units

### PR DESCRIPTION
**Pull request by pingl2016**
Correct the regex pattern for match the decimal part.

Signed-off-by: Ping Li <pingl@redhat.com>

### CODE REVIEW

**Comment by pingl2016** [comment]
ID: 1361197

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2557,19 +2557,18 @@ def __get_unit_index(M):
...
+    elif __get_unit_index(value_str[-1]) != 0:
+        unit = value_str[-1]
+        value = float(value_str[:-1])
+    elif __get_unit_index(value_str[-1]) == 0:
         raise ValueError("Invalid data size format 'value_str=%s'" % value_str)
     from_index = __get_unit_index(unit)
     to_index = __get_unit_index(order_magnitude)
     scale = int(factor) ** (to_index - from_index)
-    data_size = float(value) / scale
+    data_size = value / scale
```
Comment (PandaWei) [inline virttest/utils_misc.py]: Why remove float()?

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -2557,19 +2557,18 @@ def __get_unit_index(M):
...
 
-    regex = r"(\d+)\s*(\w?)"
-    match = re.search(regex, value_str)
-    try:
-        value = match.group(1)
-        unit = match.group(2)
-        if not unit:
-            unit = 'B'
-    except TypeError:
+    if value_str.isdigit():
```
Comment (PandaWei) [inline virttest/utils_misc.py]: Check which one, digit or decimal?
the commit message said "Get the whole data when it contains decimals ..."

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -2557,19 +2557,18 @@ def __get_unit_index(M):
             pass
         return 0
 
-    regex = r"(\d+)\s*(\w?)"
-    match = re.search(regex, value_str)
-    try:
```
Comment (PandaWei) [inline virttest/utils_misc.py]: I don't know the problems that you encounter.
IMO, Here match the value string firstly is better than call isdigit() or isdecimal()?

Comment (pingl2016) [inline virttest/utils_misc.py]: for example, if we call normalize_data_size("4.7G", "B"), the result is 4byte

Comment (PandaWei) [inline virttest/utils_misc.py]: OK, I think you can fix it by modify the regex

**Comment by pingl2016** [comment]
@PandaWei updated it, could you help to review it again?

**Comment by PandaWei** [comment]
ACK and Test

Type "help", "copyright", "credits" or "license" for more information.
>>> regex = r"(\d+\.?\d*)\s*(\w?)"
>>> value_str = "4.7G"
>>> import re
>>> match = re.search(regex, value_str)
>>> match.group(1)
'4.7'
>>> match.group(2)
'G'
>>> value_str = "4.7"
>>> match = re.search(regex, value_str)
>>> match.group(1)
'4.7'
>>> match.group(2)
''

**Comment by PandaWei** [comment]
@pingl2016 
Your latest update introduces the failure of CI

**Comment by pingl2016** [comment]
@PandaWei Correct it. thanks.

**Comment by PandaWei** [comment]
OK, Merge now

**Pull request merged**